nc/source/graphics/vulkan/MeshManager.cpp

```diff
@@ -111,8 +111,20 @@ namespace nc::graphics::vulkan
         std::vector<Vertex> allVertices = {};
         std::vector<uint32_t> allIndices = {};
         std::unordered_map<std::string, Mesh> meshes;
+        std::vector<std::string> pathsToLoad;
 
-        for (auto& path : meshPaths)
+        auto meshesExist = ResourceManager::HasMeshes();
```

Comment:
There are a few things to look at here.

The first argument to insert on line 120 was probably meant to be pathsToLoad.end()

GetMeshPaths copies the vector 3 times (at least one of them should get optimized away). ResourceManager::GetMeshPaths, ResourceManager::GetMeshPaths_, and MeshesData::GetMeshPaths should probably be returning by const& to avoid this.

Also note that you are going to copy meshPaths to pathsToLoad at some point. Either through insert on line 120 or copy initialization on 124. Because you are taking the parameter by const&, there is no other choice. If you took by value instead, you can give the caller the choice to copy or move then you can directly modify the parameter (insert, move somewhere else, whatever). This also allows you to get rid of the else statement and pathsToLoad variable entirely. If ResourceManager has meshes, you can insert those meshes directly into the meshPaths.

Reply:
Well, maybe ignore that second point. I thought MeshesData had a vector not a map.